### PR TITLE
ARROW-4694: [CI] Improve detect-changes.py on Travis PRs

### PR DESCRIPTION
**Pull request by pitrou**

### CODE REVIEW

**Comment by pitrou** [comment]
This PR is deliberately based on an old changeset.
First commit (without the fix): [many spurious changes detected](https://travis-ci.org/apache/arrow/jobs/525980681#L210)
Second commit (with the fix):
* the [right changes are detected on PR build](https://travis-ci.org/apache/arrow/jobs/525983613#L230)
* the [right changes are detected on branch build](https://travis-ci.org/pitrou/arrow/jobs/525983562#L201)

File: ci/detect-changes.py [hunk 1]
```diff
@@ -78,7 +78,17 @@ def list_travis_affected_files():
     """
     Return a list of files affected in the current Travis build.
     """
-    commit_range = get_travis_commit_range()
+    if os.environ['TRAVIS_EVENT_TYPE'] == 'pull_request':
```
Comment (fsaintjacques) [inline ci/detect-changes.py]: Can you move this logic into `get_travis_commit_range`?

**Review by fsaintjacques — CHANGES_REQUESTED** [review]
I think it's worth adding more messages with logger for future debugging, e.g. all the relevant environment variables (TRAVIS_BRANCH, TRAVIS_COMMIT, TRAVIS_COMMIT_RANGE) + applied fixes.

**Comment by pitrou** [comment]
Example debug output:
https://travis-ci.org/apache/arrow/jobs/525997684#L210-L235
https://ci.appveyor.com/project/ApacheSoftwareFoundation/arrow/builds/24176937/job/mcvlcyf9lni8bbap?fullLog=true#L9

File: ci/detect-changes.py (hunk 1 above)
Comment (nealrichardson) [reply ci/detect-changes.py]: Unless I'm misreading the diff, it looks like this logic has now been added to `get_travis_commit_range` and can be removed from here.

Comment (pitrou) [reply ci/detect-changes.py]: Er, right, it looks like my mind was distracted when doing this.

**Comment by nealrichardson** [comment]
I tried merging this into my other branch to test whether it would prevent the extra builds from being triggered, but of course `ci/detect-changes.py` is changed wrt master branch, so all of the Travis builds are running 🤦‍♂. In any case, the files that it reports as changed look correct there now, so I'd say this LGTM.

**Review by nealrichardson — APPROVED** [review]

**Pull request closed without merging**